src/main/java/store/MemoryProductStorage.java

```diff
@@ -0,0 +1,64 @@
+package store;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+public class MemoryProductStorage implements ProductStorage {
+
+	List<Produit> productList = new ArrayList<>();
+
+	@Override
+	public String save(Produit p) {
+		p.id = UUID.randomUUID().toString();
+		productList.add(p);
+		return p.id;
+	}
+
+	@Override
+	public void update(String id, Produit p) {
+		Produit product = getByID(id);
+		int productIndex = productList.indexOf(product);
+		p.id = UUID.randomUUID().toString();
+		productList.add(productIndex, p);
+		productList.remove(product);
+
+	}
+
+	@Override
+	public Produit getByID(String id) {
+		Produit p =null;
+
+		for (int i = 0; i < productList.size(); i++) {
+			if (productList.get(i).id.equals(id)) {
+				p= productList.get(i);
+			}
```

Comment:
Why not using `break` here when the desired product is found? There is no need to go through the rest of the list if the seeked product is amongst the first ones.